traffic_editor/include/traffic_editor/crowd_sim/crowd_sim_impl.h

```diff
@@ -93,15 +97,17 @@ class CrowdSimImplementation
   }
 
   void save_agent_groups(const std::vector<AgentGroup>& agent_groups);
-  std::vector<AgentGroup> get_agent_groups() const { return _agent_groups; }
+  std::vector<AgentGroup>& get_agent_groups() { return _agent_groups; }
```

Comment:
This change is a bit suspicious, if a user was to edit the returned variable it would also affect the state of the class, it's not quite the behavior I would expect from a getter function.

Reply:
it is intended to be both get and set (editable and mutable). Not sure if simply setting it to an accessible public property can do the job (i.e. remove the getter/setter function).

Reply:
It seems to me that's what the [save_agent_groups](https://github.com/osrf/traffic_editor/blob/d7646a200e7d55078089640a9883e8d343618a42/traffic_editor/gui/crowd_sim/crowd_sim_impl.cpp#L367) function did, so you would get the agent_groups, edit them and then `save` (really `set`) in a subsequent call.